### PR TITLE
categories-smaller-steps, component

### PR DESCRIPTION
**Pull request by frunolu**

### CODE REVIEW

**Comment by frunolu** [comment]
DONE

File: project/app/Presenters/BasePresenter.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+namespace App\Presenters;
+
+use App\Components\CategoriesMenu\CategoriesMenu;
+use Nette\Application\UI\Presenter;
+
+class BasePresenter extends Presenter
+{
+    /**
+     * @return CategoriesMenu
+     */
```
Comment (jendo) [inline project/app/Presenters/BasePresenter.php:14]: Zbytocna anotacia lebo to iste uz mam popisane v ramci return type

File: project/app/Components/CategoriesMenu/CategoriesMenu.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Components\CategoriesMenu;
+
+use Nette\Application\UI\Control;
+
+class CategoriesMenu extends Control
```
Comment (jendo) [inline project/app/Components/CategoriesMenu/CategoriesMenu.php:9]: Komponentu by som premenoval na MainMenu (CategoriesMenu mi nedava zmysel)

File: project/app/Presenters/BasePresenter.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            [
+                'url' => 'pekarna-a-cukrarna',
+                'name' => 'Pekárna a cukrárna'
+            ],
+            [
+                'url' => 'maso-a-ryby',
+                'name' => 'Maso a ryby'
+            ]
+        ];
+        return new CategoriesMenu($links);
```
Comment (jendo) [inline project/app/Presenters/BasePresenter.php:27]: PSR12

File: project/app/Components/CategoriesMenu/CategoriesMenu.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    public function render(): void
+    {
+        $this->getTemplate ()->setFile (__DIR__ . '/templates/CategoriesMenu.latte');
+        $this->getTemplate ()->links = $this->links;
+        $this->getTemplate ()->render ();
+    }
+
+    public function addLink($url, $name): void
```
Comment (jendo) [inline project/app/Components/CategoriesMenu/CategoriesMenu.php:30]: typova kontrola

File: project/app/Components/CategoriesMenu/CategoriesMenu.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<?php
+
```
Comment (jendo) [inline project/app/Components/CategoriesMenu/CategoriesMenu.php:2]: PSR12

File: project/app/Components/CategoriesMenu/CategoriesMenu.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+use Nette\Application\UI\Control;
+
+class CategoriesMenu extends Control
+{
+    /**
+     * @var array
+     */
+    private array $links = [];
+
+    public function __construct(array $links = [])
```
Comment (jendo) [inline project/app/Components/CategoriesMenu/CategoriesMenu.php:16]: Defaultne pole pre hlavne menu nedava zmysel

File: project/app/Presenters/BasePresenter.php (hunk 1 above)
Comment (frunolu) [reply project/app/Presenters/BasePresenter.php]: DONE

File: project/app/Components/CategoriesMenu/CategoriesMenu.php (hunk 2 above)
Comment (frunolu) [reply project/app/Components/CategoriesMenu/CategoriesMenu.php]: DONE

File: project/app/Presenters/BasePresenter.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        $links = [
+            [
+                'url' => 'pekarna-a-cukrarna',
+                'name' => 'Pekárna a cukrárna'
+            ],
+            [
+                'url' => 'maso-a-ryby',
+                'name' => 'Maso a ryby'
+            ]
```
[the remark this answers is not in the document]
Comment (frunolu) [reply project/app/Presenters/BasePresenter.php]: DONE

File: project/app/Presenters/BasePresenter.php (hunk 3 above)
Comment (frunolu) [reply project/app/Presenters/BasePresenter.php]: DONE

File: project/app/Components/CategoriesMenu/CategoriesMenu.php (hunk 4 above)
Comment (frunolu) [reply project/app/Components/CategoriesMenu/CategoriesMenu.php]: DONE

**Pull request merged**